solvers/glmnet.py

```diff
@@ -38,18 +41,18 @@ def set_objective(self, X, y, lmbd, fit_intercept):
         self.X, self.y, self.lmbd = X, y, lmbd
         self.fit_intercept = fit_intercept
 
-        self.lmbd_max = np.max(np.abs(X.T @ y))
         self.glmnet = robjects.r['glmnet']
 
-    def run(self, n_iter):
-        fit_dict = {"lambda.min.ratio": self.lmbd / self.lmbd_max}
+    def run(self, tol):
+        fit_dict = {"lambda": self.lmbd / len(self.y)}
+
         glmnet_fit = self.glmnet(self.X, self.y, intercept=False,
-                                 standardize=False, maxit=n_iter,
-                                 thresh=1e-14, **fit_dict)
+                                 standardize=False, maxit=1_000_000,
+                                 thresh=tol / len(self.y), **fit_dict)
```

Comment:
From GLMNET doc, `thresh` is: 
Convergence threshold for coordinate descent. Each inner coordinate-descent loop continues until the relative change in any coefficient is less than thresh. Defaults value is 1E-4.

@jolars Do you know how we could scale it so that it's robust across datasets ? In particular, from this definition, it's not clear to me why we scale it by n_samples.

Reply:
Yeah, the `n_samples` thing was just taken out of the air. 

Actually, I'm not sure we really need to scale it at all. The relative change in coefficients should not be sensitive to scaling or dimensions. Maybe something like `thresh = tol^c` where c is some constant is more reasonable. Maybe `c = 2`? I'll experiment a bit.

Reply:
By that I mean, not scale it according to the dimensions. We just need to apply some general modification.

Reply:
It's a great idea to put tol ** 2 (from what we saw yesterday you sometimes need to go down to really low values of thresh, like `1e-20`, so the square will help because benchopt's policy is much less agressive.

You're right for the scaling, it's a relative change in coef (I'm thinking norm(w_{t+1} - w_t / w_t, ond=np.inf} <= thresh) so it should be adimensional.

Reply:
With c=2 I still got a stuck solver before convergence. c= 3 is wild for one config:
![image](https://user-images.githubusercontent.com/8993218/145032853-65d2dc2b-75f5-45a3-8ebf-e72d147d0636.png)

(how can it not converge on such a simple problem ? lambda is very large and max_iter is huge!) 


c = 2.3 seems ok on these datasets